### ISSUE TITLE
Printing of an empty UnitRange at the REPL lacks information

### ISSUE THREAD

**Issue by rfourquet** [JuliaLang/julia#13966]
An empty `UnitRange` holds more information than the fact that it's empty, this is used with e.g. `searchsorted` to indicate the "insertion point". The following session leaves to be desired:
```
julia> searchsorted([1, 3], 2)
0-element UnitRange{Int64}

julia> ans.start
2
```
Showing instead something like "0-element UnitRange{Int64} 2:1" would help. Actually, I would personally prefer that even a non-empty range is shown like this in compact form (before the list of its values).

I guess this was made in #13615, cc @artkuo .

**Comment by nalimilan** [comment]
+1 to also showing the compact form on the first line. After all, that's the only way to recreate the same object via copy/paste at the REPL.

**Comment by StefanKarpinski** [comment]
I really hate the new range printing. Anyone have a link to that discussion and PR handy?

**Comment by yuyichao** [comment]
It was in the first post https://github.com/JuliaLang/julia/pull/13615 (the PR also links the julia-user discussions)

**Comment by artkuo** [comment]
That was me, in response to complaints that people wanted to see what's actually in the `Range`. I agree there's useful info that could be included in first line. What would you say to something like this? 

    julia> 1:3
    3-element UnitRange{Int64}: 1:3
     1,2,3

Is there any issue with the colons being so close to each other? The standard summary usually has a colon, as does a `StepRange` or `UnitRange`.

**Comment by StefanKarpinski** [comment]
I think we should print ranges the way we've always printed them. LinSpace can get special treatment. Ot we can just go back to returning an array, I'm really tired of complaints about this.

**Comment by artkuo** [comment]
Did you have something specific in mind about how to return an array and still convey `Range`? The new method was trying to address that. And the previous way of only showing `1:3` or `linspace(1,3,3)` was what caused a lot of complaints, as people were expecting something more.

A nice thing about `AbstractVector` is that clever algebras can be implemented, but the uninterested user can remain relatively naive to the underpinnings. At the REPL, the average user may want to see what's in the `Range`, while some may want to see the definition. I think @rfourquet is right to provide useful info both ways. My question is how to format it, maybe with a comma?

    julia> 1:3
    3-element UnitRange{Int64} 1:3,
     1,2,3
    
    julia> 3:2
    0-element UnitRange{Int64) 3:2

Is there a mechanism to have REPL options available for the user to set? The equivalent of "format long" in Matlab? REPL behavior can be highly personal, and it might be nice to offer a way for a person to choose, if it doesn't exist already.

**Comment by StefanKarpinski** [comment]
Printing `1:3` never received a single complaint, it was only the `linspace` thing that people don't like. I'm sick of this and ready to just make `linspace` return and array and get rid of the `LinSpace` type.

**Comment by artkuo** [comment]
@StefanKarpinski I think the `LinSpace` complaints will go away when people see the REPL behavior return what they expect. Otherwise, it should be relatively transparent to the user with lots of potential benefits. What we are proposing here is to have our cake and eat it too. I'll try to file a PR to give something concrete to discuss and either accept or reject.

**Comment by StefanKarpinski** [comment]
How about leaving other ranges the way they were and just printing `LinSpace` objects like vectors?

**Comment by nalimilan** [comment]
Wouldn't that be too inconsistent, only to reduce the surprise for  users coming from MATLAB? The different treatment of `LinSpace` and `FloatRange` is particularly hard to justify IMHO.

**Comment by StefanKarpinski** [comment]
The justification would seem to be that people are freaked out by one and unphased by the other.

**Comment by artkuo** [comment]
For reference here's the 0.4 behavior:

    julia> 1:3
    1:3

    julia> 3:2
    3:2

    julia> linspace(1,3,3)
    linspace(1.0,3.0,3)

Here's the [discussion](https://groups.google.com/forum/#!searchin/julia-users/range/julia-users/qPqgJS-usrU/TVSuIygPCAAJ) that started this.Why not show more info like number of elements, element type, what's inside, etc.?

**Comment by JeffBezanson** [comment]
fixed by #15799
